connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1020,7 +1107,9 @@ public HerderRequest(long at, Callable<Void> action, Callback<Void> callback) {
 
         @Override
         public int compareTo(HerderRequest o) {
-            return Long.compare(at, o.at);
+            final int soonest = Long.compare(at, o.at);
+            // If tied, returning a positive value should respect insertion order.
+            return soonest != 0 ? soonest : 1;
```

Comment:
> If tied, returning a positive value should respect insertion order.

that's a bit weird... maybe better to just have a counter used for breaking ties?